### ISSUE TITLE
Improve a11y via keyboard navigable NavBar and "Skip to Content Button"

### ISSUE THREAD

**Issue by andrico1234** [marmelab/react-admin#5764]
**Is your feature request related to a problem? Please describe.**
I appreciate how easy to use React Admin works out of the box in regards to keyboard navigation in the Edit/Show/Create/List screens, and with just a little work the same can be achieved in the other parts of the library

If you're happy to have these included in the app, I would love to create the MR(s).

The two proposals I have are

- Left side navigation accepts focus: https://github.com/marmelab/react-admin/pull/5772
- Add a "Skip to Content button": https://github.com/marmelab/react-admin/pull/5773

The former will ensure that all buttons and links in the page are tabbable, since it seems that RA skips over the list of resources in the side menu.

The latter will allow users to jump straight to the main content on the page.

These are ARIA standards outlined in: [here](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_general_between) and [here](https://www.w3.org/TR/WCAG20-TECHS/G1.html)

### Focusable Nav Bar

It looks like the dashboard items have a `tabIndex` set to -1. Removing this makes the button tabbable.

![image](https://user-images.githubusercontent.com/17087167/104438627-19b2e600-5588-11eb-9b75-9aff1e252ecf.png) 

### Skip to content button

![image](https://user-images.githubusercontent.com/17087167/104438742-3d762c00-5588-11eb-8b49-99ebde0f7fc7.png)

This should be relatively straight forward, this would include a button like the following

```jsx
function SkipNavigation() {
  return (
    <a className="SkipNavigation" tabIndex={0} href="#main">
      Skip Navigation
    </a>
  );
}
```

```css
.SkipNavigation {
  top: -200px;
  left: 0;
  position: absolute;
}

.SkipNavigation:focus {
  top: 0;
}
```

And then adding an id of `main` to either the outer-most `<div>` tag in files like [this](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/detail/EditView.tsx), or a better place if it allows us to make changes in the fewest number of places.

**Comment by fzaninotto** [comment]
Great idea! Would you like to work on a PR to implement this (on the `next` branch)?

**Comment by fzaninotto** [comment]
Fixed by #5772 and #5804
